File: example_fragment_proxy_activity_result/src/main/java/com/simple/example_fragment_proxy_activity_result/MainActivity.java
package com.simple.example_fragment_proxy_activity_result;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }

    public void onClick(View view) {

       ActivityResultProxy.with(this)
                .setToActivity(ToActivity.class)
                .setRequestCode(100)
                .startActivityForResult(new OnResultListener() {
                    @Override
                    public void onActivityResult(int requestCode, int resultCode, Intent data) {

                        Log.e("simple", "requestCode ==" + requestCode);
                        Log.e("simple", "resultCode ==" + resultCode);
                        Log.e("simple", "data ==" + data);
                    }
                })
                .start();

    }

//    @Override
//    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
//        super.onActivityResult(requestCode, resultCode, data);
//        Log.e("simple", "requestCode ==" + requestCode);
//        Log.e("simple", "resultCode ==" + resultCode);
//        Log.e("simple", "data ==" + data);
//    }
}

File: example_activity_lifecycle_export/src/main/java/com/example/activity_lifecycle_export/MainActivity.kt
package com.example.activity_lifecycle_export

import android.content.ComponentName
import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tv_export.setOnClickListener {
            //第一个参数是Activity所在的package包名，第二个参数是完整的Class类名（包括包路径）
            val componetName =  ComponentName("com.example.activity_lifecycle",
                    "com.example.activity_lifecycle.MainActivity")
            val intent =  Intent()
            intent.setComponent(componetName);
            startActivity(intent);
        }
    }
}
